[CI] TSDBIndexingIT testTrimId failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3845 / amazonlinux-2_platform-support-aws](https://gradle-enterprise.elastic.co/s/5t563ufoccpjs)
- [elasticsearch-periodic-platform-support #3842 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/c2et2sohjqdia)
- [elasticsearch-periodic-platform-support #3830 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/t63ido67pjljg)
- [elasticsearch-periodic-platform-support #3827 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/4dwxmw6lyodbc)
- [elasticsearch-periodic-platform-support #3824 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/mxqmsjwl5wtmm)
- [elasticsearch-periodic-platform-support #3821 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/kq2wnpxpjufcy)
- [elasticsearch-periodic-platform-support #3818 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/nut4hldhgfkho)

**Reproduction Line:**
```
./gradlew ":modules:data-streams:internalClusterTest" --tests "org.elasticsearch.datastreams.TSDBIndexingIT.testTrimId" -Dtests.seed=8E2D592633C80A88 -Dtests.locale=sk -Dtests.timezone=Asia/Urumqi -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testTrimId'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.datastreams.TSDBIndexingIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a collection with size a value equal to or greater than <2>
     but: collection size <1> was less than <2>
```

**Issue Reasons:**
- [main] 7 failures in test testTrimId (1.9% fail rate in 364 executions)
- [main] 6 failures in step amazonlinux-2023_platform-support-aws (66.7% fail rate in 9 executions)
- [main] 7 failures in pipeline elasticsearch-periodic-platform-support (70.0% fail rate in 10 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBIndexingIT.java
+++ b/modules/data-streams/src/internalClusterTest/java/org/elasticsearch/datastreams/TSDBIndexingIT.java
@@ -22,6 +22,7 @@ import org.elasticsearch.action.bulk.BulkRequest;
 import org.elasticsearch.action.get.GetRequest;
 import org.elasticsearch.action.index.IndexRequest;
 import org.elasticsearch.action.search.SearchRequest;
+import org.elasticsearch.action.support.WriteRequest;
 import org.elasticsearch.cluster.metadata.ComponentTemplate;
 import org.elasticsearch.cluster.metadata.ComposableIndexTemplate;
 import org.elasticsearch.cluster.metadata.Template;
@@ -35,6 +36,7 @@ import org.elasticsearch.index.IndexSettings;
 import org.elasticsearch.index.query.RangeQueryBuilder;
 import org.elasticsearch.indices.InvalidIndexTemplateException;
 import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.rest.RestStatus;
 import org.elasticsearch.search.builder.SearchSourceBuilder;
 import org.elasticsearch.test.ESSingleNodeTestCase;
 import org.elasticsearch.test.InternalSettingsPlugin;
@@ -457,6 +459,16 @@ public class TSDBIndexingIT extends ESSingleNodeTestCase {
                 indexName = bulkResponse.getItems()[0].getIndex();
             }
             client().admin().indices().refresh(new RefreshRequest(dataStreamName)).actionGet();
+
+            // In rare cases we can end up with a single segment shard, which means we can't trim away the _id later.
+            // So update an existing doc to create a new segment without adding a new document after force merging:
+            var indexRequest = new IndexRequest(indexName).setIfPrimaryTerm(1L)
+                .setIfSeqNo((numBulkRequests * numDocsPerBulk) - 1)
+                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+            indexRequest.source(DOC.replace("$time", formatInstant(time.minusMillis(1))), XContentType.JSON);
+            var res = client().index(indexRequest).actionGet();
+            assertThat(res.status(), equalTo(RestStatus.OK));
+            assertThat(res.getVersion(), equalTo(2L));
         }
 
         // Check whether there are multiple segments:
@@ -494,7 +506,7 @@ public class TSDBIndexingIT extends ESSingleNodeTestCase {
             assertThat(retentionLeasesStats.retentionLeases().leases(), hasSize(1));
             assertThat(
                 retentionLeasesStats.retentionLeases().leases().iterator().next().retainingSequenceNumber(),
-                equalTo((long) numBulkRequests * numDocsPerBulk)
+                equalTo((long) numBulkRequests * numDocsPerBulk + 1)
             );
         });
 

